packages/vue-components/src/panels/PanelBase.js

```diff
@@ -95,80 +95,133 @@ export default {
+
+      this.localExpanded = !this.localExpanded;
     },
     close() {
-      this.localExpanded = false;
       this.localMinimized = true;
+      this.localExpanded = false;
+      /*
+        We do not need transition for closing panels (changing to minimized).
+        Thus, we do not use nextTick here.
+      */
+      this.$refs.panel.style.maxHeight = `${this.collapsedPanelHeight}px`;
     },
     open() {
+      this.localMinimized = false;
+      this.localExpanded = true;
       this.wasRetrieverLoaded = true;
+      /*
+        After setting minimized to false / wasRetrieverLoaded to true,
+        we have to wait for DOM update (nextTick) before setting maxHeight for transition.
+      */
       this.$nextTick(() => {
-        this.localExpanded = true;
+        this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
       });
-      this.localMinimized = false;
     },
     openPopup() {
       window.open(this.popupUrl);
     },
-    setMaxHeight() {
-      // Don't play the transition for this case as the loading should feel 'instant'
+    retrieverUpdateMaxHeight() {
+      // src has finished loaded -- we set this flag to true so our event listener can set maxHeight to none
+      this.isRetrieverLoadDone = true;
+
+      if (this.preloadBool && !this.wasRetrieverLoaded) {
+        // Only preload, do not expand the panel.
+        return;
+      }
+      // Don't play the transition for this case as the loading should feel 'instant'.
       if (this.expandedBool) {
         this.$refs.panel.style.maxHeight = 'none';
         return;
       }
 
-      /*
-      Otherwise, since the vue transition is dependent on localExpanded, we have to manually
-      set our own transition end handlers here for the initial loading of the content.
-      */
-      const onExpandDone = () => {
-        this.$refs.panel.style.maxHeight = 'none';
-        this.$refs.panel.removeEventListener('transitionend', onExpandDone);
-      };
-
-      this.$refs.panel.addEventListener('transitionend', onExpandDone);
+      // For expansion transition to 'continue' after src is loaded.
       this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
     },
-    beforeExpand(el) {
-      el.style.maxHeight = '0';
-    },
-    duringExpand(el) {
-      jQuery('html').stop();
-      el.style.maxHeight = `${el.scrollHeight}px`;
-    },
-    afterExpand(el) {
-      el.style.maxHeight = 'none';
-    },
-    beforeCollapse(el) {
-      el.style.maxHeight = `${el.scrollHeight}px`;
+    initCollapsedPanelHeight() {
+      // To set panel's default collapsed panel height.
+      this.collapsedPanelHeight = 0;
     },
-    duringCollapse(el) {
-      if (this.$el.getBoundingClientRect().top < 0) {
-        jQuery('html').animate({
-          scrollTop: window.scrollY + this.$el.getBoundingClientRect().top - 3,
-        }, 500, 'swing');
+    initPanel() {
+      this.$refs.panel.addEventListener('transitionend', () => {
+        /*
```

Comment:
some suggestions to shorten this comment:

Reply:
Thanks for the suggestion! I agree that it was rather lengthy :-)